src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js

```diff
@@ -13,24 +13,22 @@ import TwoFactorAuthContext from './TwoFactorAuthContext';
 import {defaultAccount, TwoFactorAuthPropTypes} from './TwoFactorAuthPropTypes';
 
 function TwoFactorAuthSteps({account = defaultAccount}) {
-    const [currentStep, setCurrentStep] = useState(CONST.TWO_FACTOR_AUTH_STEPS.CODES);
+    const calculateCurrentStep = useMemo(() => {
```

Comment:
There is some weirdness going on in this component, essentially derived state. At minimum, I'd rename `calculateCurrentStep` to `currentStepDerivedFromProps` and move the related `useEffect` up:

```diff
diff --git a/scripts/start-android.sh b/scripts/start-android.sh
old mode 100644
new mode 100755
diff --git a/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js b/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js
index 116d44e9e3..0dfc6ef951 100644
--- a/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js
+++ b/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js
@@ -13,23 +13,21 @@ import TwoFactorAuthContext from './TwoFactorAuthContext';
 import {defaultAccount, TwoFactorAuthPropTypes} from './TwoFactorAuthPropTypes';
 
 function TwoFactorAuthSteps({account = defaultAccount}) {
-    const calculateCurrentStep = useMemo(() => {
+    const currentStepDerivedFromProps = useMemo(() => {
         if (account.twoFactorAuthStep) {
             return account.twoFactorAuthStep;
         }
         return account.requiresTwoFactorAuth ? CONST.TWO_FACTOR_AUTH_STEPS.ENABLED : CONST.TWO_FACTOR_AUTH_STEPS.CODES;
     }, [account.requiresTwoFactorAuth, account.twoFactorAuthStep]);
-
-    const [currentStep, setCurrentStep] = useState(calculateCurrentStep);
+    const [currentStep, setCurrentStep] = useState(currentStepDerivedFromProps);
+    useEffect(() => {
+        setCurrentStep(currentStepDerivedFromProps);
+    }, [currentStepDerivedFromProps]);
 
     const {setAnimationDirection} = useAnimatedStepContext();
 
     useEffect(() => () => TwoFactorAuthActions.clearTwoFactorAuthData(), []);
 
-    useEffect(() => {
-        setCurrentStep(calculateCurrentStep);
-    }, [calculateCurrentStep]);
-
     const handleSetStep = useCallback(
         (step, animationDirection = CONST.ANIMATION_DIRECTION.IN) => {
             setAnimationDirection(animationDirection);
```

However, I think a simpler approach would be to just use props and rely on Onyx to perform the update:

```diff
diff --git a/scripts/start-android.sh b/scripts/start-android.sh
old mode 100644
new mode 100755
diff --git a/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js b/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js
index 116d44e9e3..4d342c8dac 100644
--- a/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js
+++ b/src/pages/settings/Security/TwoFactorAuth/TwoFactorAuthSteps.js
@@ -13,28 +13,22 @@ import TwoFactorAuthContext from './TwoFactorAuthContext';
 import {defaultAccount, TwoFactorAuthPropTypes} from './TwoFactorAuthPropTypes';
 
 function TwoFactorAuthSteps({account = defaultAccount}) {
-    const calculateCurrentStep = useMemo(() => {
+    const currentStep = useMemo(() => {
         if (account.twoFactorAuthStep) {
             return account.twoFactorAuthStep;
         }
         return account.requiresTwoFactorAuth ? CONST.TWO_FACTOR_AUTH_STEPS.ENABLED : CONST.TWO_FACTOR_AUTH_STEPS.CODES;
     }, [account.requiresTwoFactorAuth, account.twoFactorAuthStep]);
 
-    const [currentStep, setCurrentStep] = useState(calculateCurrentStep);
-
     const {setAnimationDirection} = useAnimatedStepContext();
 
     useEffect(() => () => TwoFactorAuthActions.clearTwoFactorAuthData(), []);
 
-    useEffect(() => {
-        setCurrentStep(calculateCurrentStep);
-    }, [calculateCurrentStep]);
-
     const handleSetStep = useCallback(
         (step, animationDirection = CONST.ANIMATION_DIRECTION.IN) => {
             setAnimationDirection(animationDirection);
             TwoFactorAuthActions.setTwoFactorAuthStep(step);
-            setCurrentStep(step);
         },
         [setAnimationDirection],
     );
```

Note that `TwoFactorAuthActions.setTwoFactorAuthStep` will immediately call `Onyx.merge` with the new step, then thanks to `withOnyx` the component will re-render with the new `twoFactorAuthStep` prop, and `currentStep` will be recomputed accordingly.